README.md

```diff
@@ -171,11 +171,36 @@ The shared storage worklet invocation methods (`addModule`, `run`, and `selectUR
     *   Creates a new worklet, and loads and adds the module to the worklet (similar to the handling for `window.sharedStorage.worklet.addModule(url, options)`).
     *   By default, the worklet uses the invoking context's origin as its partition origin for accessing shared storage data and for budget checking and withdrawing.
         *   To instead use the worklet script origin (i.e. `url`'s origin) as the partition origin for accessing shared storage, pass the `dataOrigin` option with "script-origin" as its value in the `options` dictionary.
-        *   Currently, the `dataOrigin` option, if used, is restricted to having either "script-origin" or "context-origin" as its value. "script-origin" designates the worklet script origin as the data partition origin; "context-origin" designates the invoking context origin as the data partition origin.
+        *   To use a custom origin as the partition origin for accessing shared storage, pass the `dataOrigin` option with the serialized partition origin (i.e. the [URL serialization](https://url.spec.whatwg.org/#url-serializing) of a URL with the same scheme, host, and port as the partition origin, but whose other components are empty) as its value in the `options` dictionary.
+        *   Supported values for the `dataOrigin` option, if used, are the keywords "script-origin" and "context-origin", as well as any valid serialized HTTPS origin, e.g. "https://custom-data-origin.example". 
+            *   "script-origin" designates the worklet script origin as the data partition origin.
+            *   "context-origin" (default) designates the invoking context origin as the data partition origin.
+            *   If a valid serialized HTTPS URL is passed as the value for `dataOrigin`, then the origin of the URL parsed from the `dataOrigin` string will be used as the worklet's data partition origin, as long as the URL's origin has opted-in via a [/.well-known/](#well-known) to specify that the worklet script's origin and context origin are trusted.
```

Comment:
Suggest to keep it brief here, e.g. "a serialized HTTPS origin designates itself as the data partition origin." And then, right next to this paragraph, add the paragraph "When a valid serialized HTTPS URL is passed as the value for dataOrigin ..", which explains how the data origin is treated. This removes some of the redundancy.

Reply:
Done.